### PR TITLE
refactor: Extract `AppVersionText` into a separate widget + Add a new widget = `ChipText` 

### PR DESCRIPTION
**Pull request by wal33d006**

### CODE REVIEW

**Review by Zfinix — APPROVED** [review]

File: packages/cosmos_ui_components/lib/components/app_version_text.dart [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import 'package:cosmos_utils/app_info_provider.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+class AppVersionText extends StatelessWidget {
+  const AppVersionText({required this.appInfoProvider, Key? key}) : super(key: key);
```
Comment (andrzejchm) [inline packages/cosmos_ui_components/lib/components/app_version_text.dart:6]:
```suggestion
  const AppVersionText({required this.appInfoProvider, Key? key,}) : super(key: key);
```

File: packages/cosmos_ui_components/lib/components/app_version_text.dart [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  final AppInfoProvider appInfoProvider;
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
```
Comment (andrzejchm) [inline packages/cosmos_ui_components/lib/components/app_version_text.dart:10]: this method should be at the bottom of the class

File: packages/cosmos_ui_components/lib/components/app_version_text.dart [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty<AppInfoProvider>('appInfoProvider', appInfoProvider));
+  }
+
+  @override
+  Widget build(BuildContext context) => FutureBuilder<String>(
+        future: appInfoProvider.getAppVersion(),
+        builder: (context, snapshot) => Text(snapshot.data ?? ''),
```
Comment (andrzejchm) [inline packages/cosmos_ui_components/lib/components/app_version_text.dart:18]: right now we're not able to style it, let's add another optional `TextStyle?` to the constructor of `AppVersionText` and pass it into this `Text` widget

File: packages/cosmos_ui_components/lib/components/chip_text.dart [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  final String title;
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = CosmosTheme.of(context);
+    return Container(
+      padding: EdgeInsets.all(theme.spacingM),
+      decoration: BoxDecoration(
+        borderRadius: BorderRadius.circular(10),
+        color: theme.colors.chipBackground,
```
Comment (andrzejchm) [inline packages/cosmos_ui_components/lib/components/chip_text.dart:19]: let's make the `Color? color` an optional parameter to the `ChipText` widget with the default value of `theme.colors.chipBackground`

File: packages/cosmos_ui_components/lib/components/chip_text.dart [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      ),
+      child: Text(title),
```
Comment (andrzejchm) [inline packages/cosmos_ui_components/lib/components/chip_text.dart:21]: let's add optional parameter of `TextStyle?` to the `ChipText` constructor and pass it to this `Text` widget

**Review by andrzejchm — CHANGES_REQUESTED** [review]
looks great! lets just add some optional styling capabilities to the widgets

**Review by andrzejchm — APPROVED** [review]
🌟

**Pull request merged**